### PR TITLE
Add --host_action_env option

### PR DESCRIPTION
**Pull request by mai93**
This PR adds `--host_action_env` option to specify a set of environment variables to be added to the host configuration. The variables specified by this option are only added to host configuration while those specified by `--action_env` are only included in the target configuration.

Fixes: #4008

### CODE REVIEW

**Comment by mai93** [comment]
cc @philwo @meteorcloudy

File: src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java [hunk 1]
```diff
@@ -342,14 +342,30 @@ public String getTypeDescription() {
       documentationCategory = OptionDocumentationCategory.OUTPUT_PARAMETERS,
       effectTags = {OptionEffectTag.ACTION_COMMAND_LINES},
       help =
-          "Specifies the set of environment variables available to actions. "
+          "Specifies the set of environment variables available to actions performed using target configuration. "
```
Comment (meteorcloudy) [inline src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:345]: Maybe "with target configuration" sounds a bit simper? Same below.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java (hunk 1 above)
Comment (mai93) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:345]: To be "available to actions with target configuration." ?

Comment (meteorcloudy) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:345]: Yes, that's what I mean

**Comment by gregestren** [comment]
I'm going to punt this to @katre. Eventually I think we should move away from needing explicit `--foo` / `--host_foo` flags toward a real API for specifying which flags go to the host and which don't. @katre had some ideas about evolving these themes.

I also acknowledge we won't have that tomorrow. So the above isn't necessarily an objection to this approach.

File: src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java [hunk 2]
```diff
@@ -342,14 +342,30 @@ public String getTypeDescription() {
...
 
+  @Option(
+          name = "host_action_env",
+          converter = Converters.OptionalAssignmentConverter.class,
+          allowMultiple = true,
+          defaultValue = "null",
+          documentationCategory = OptionDocumentationCategory.OUTPUT_PARAMETERS,
+          effectTags = {OptionEffectTag.ACTION_COMMAND_LINES},
+          help =
+                  "Specifies the set of environment variables available to actions with host configuration. "
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:361]: "with host or execution configurations".

Ideally we'd move away from target/host to non-tool/tool, but we're not there now so no need to burden your PR.

**Review by katre — APPROVED** [review]
This looks fine with a small fix to the doc comment.

**Comment by katre** [comment]
Currently, host.actionEnvironment is left unset, correct? Then no need to start propogating the target actionEnvironment to the host configuration.

**Comment by meteorcloudy** [comment]
Thanks for the clarification! Mai, can you fix the documentation? Then you can import this change.

**Review by katre — APPROVED** [review]

**Pull request closed without merging**